docs/src/main/sphinx/security/oauth2.rst

```diff
@@ -0,0 +1,106 @@
+========================
+OAuth 2.0 authentication
+========================
+
+Trino can be configured to enable OAuth 2.0 authentication over HTTPS for the
+Web UI and the JDBC driver. Trino uses the `Authorization Code
+<https://tools.ietf.org/html/rfc6749#section-1.3.1>`_ flow which exchanges an
+Authorization Code for a token. At a high level, the flow includes the following
+steps:
+
+#. the Trino coordinator redirects a user agent to the Authorization Server
+#. the user authenticates with the Authorization Server, and it approves the Trino's permissions request
+#. the user agent is redirected back to the Trino coordinator with an authorization code
+#. the Trino coordinator exchanges the authorization code for a token
+
+To enable OAuth 2.0 authentication for Trino, configuration changes are made on
+the Trino coordinator. No changes are required to the worker configuration;
+only the communication from the clients to the coordinator is authenticated.
+
+Trino server configuration
+--------------------------
+
+Using the OAuth2 authentication requires the Trino coordinator to be secured
+with TLS.
+
+The following is an example of the required properties that need to be added
+to the coordinator's ``config.properties`` file:
+
+.. code-block:: properties
+
+    http-server.authentication.type=oauth2
+
+    http-server.https.port=8443
+    http-server.https.enabled=true
+
+    http-server.authentication.oauth2.auth-url=https://authorization-server.com/authorize
+    http-server.authentication.oauth2.token-url=https://authorization-server.com/token
+    http-server.authentication.oauth2.jwks-url=https://authorization-server.com/.well-known/jwks.json
+    http-server.authentication.oauth2.client-id=CLIENT_ID
+    http-server.authentication.oauth2.client-secret=CLIENT_SECRET
+
+In order to enable OAuth 2.0 authentication for the Web UI, the following
+properties need to be added:
+
+.. code-block:: properties
+
+    web-ui.enabled=true
+    web-ui.authentication.type=oauth2
```

Comment:
This is also the default, unless you have multiple authenticators installed

Reply:
Should I remove it or change the description?

Reply:
I don't know, I was just pointting this out.  @mosabua what do you think?

Reply:
I think we should leave it .. less hidden magic.